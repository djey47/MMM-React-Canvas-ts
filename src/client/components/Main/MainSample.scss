/** @define main-sample */

/**
 * MagicMirror
 * Style rules for Main component
 */

.main-sample {
  font-weight: bold;

  &__description {
    font-style: italic;
  }
}
