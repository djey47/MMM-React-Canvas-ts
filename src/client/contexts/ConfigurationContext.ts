import { createContext } from 'react';

const ConfigurationContext = createContext<ModuleConfiguration | undefined>({});
export default ConfigurationContext;
