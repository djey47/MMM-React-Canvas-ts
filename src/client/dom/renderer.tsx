import ReactDOM from 'react-dom';

import MainSample from '../components/Main/MainSample';

/**
 * MagicMirror
 * HTML renderer
 */

/**
 * @returns HTML for main wrapper
 */
export const renderWrapper = (wrapperId: string): HTMLDivElement => {
  const wrapper = document.createElement('div');
  wrapper.id = wrapperId;
  wrapper.className = wrapperId;

  return wrapper;
};

/**
 * REACT gateway helper
 * @return Mounted component
 */
export const renderMainComponent = (wrapperId: string): void => {
  const rootId = document.getElementById(wrapperId);
  if (!rootId) {
    Log.error(`** Could not find root div with id: ${wrapperId}! Aborting.`);
    return;
  }

  ReactDOM.render(
    <MainSample />,
    rootId
  );
};
