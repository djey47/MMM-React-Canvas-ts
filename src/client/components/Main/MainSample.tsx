import { PureComponent, useContext } from 'react';
import classnames from 'classnames';
import Subscribed from '../Subscribed/Subscribed';

import '../../styles/reset.scss';
import '../../styles/module.scss';
import './MainSample.scss';
import ConfigurationContext from '../../contexts/ConfigurationContext';

interface Props {
  oneProp?: string;
}

interface State {
  oneStateProperty?: number;
}

/**
 * MagicMirror
 * Main REACT component sample
 * Rename this component and associated resources to your convenience
 */
class MainSample extends PureComponent<Props, State> {
  static defaultProps = {};

  constructor(props: Props) {
    super(props);
  }

  /**
   * @return component markup
   */
  render() {
    return (
      <ConfigurationContext.Consumer>
        {(configuration) => (
          <div className={classnames('main-sample', 'dimmed', 'light', 'small')}>
            <p className="main-sample__description">
              Main component sample for MM2's React-Typescript template
            </p>
            <p className="main-sample__configuration">
              Configuration in context:
              <pre>
                <code>
                  {JSON.stringify(configuration)}
                </code>
              </pre>
            </p>
            <Subscribed prop1="text" />
          </div>
        )}
      </ConfigurationContext.Consumer>
    );
  }
}

export default MainSample;
