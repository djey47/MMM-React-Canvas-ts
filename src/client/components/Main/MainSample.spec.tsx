import renderer from 'react-test-renderer';
import ConfigurationContext from '../../contexts/ConfigurationContext';
import MainSample from './MainSample';

jest.mock('../Subscribed/Subscribed', () => (props: object) => (
  <div {...props}>Subscribed component</div>
));

describe('MainSample component', () => {
  it('should render correctly with default props', () => {
    const tree = renderer.create(<MainSample />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with configuration context', () => {
    const tree = renderer.create(
      <ConfigurationContext.Provider value={{debug: true}}>
        <MainSample />
      </ConfigurationContext.Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
