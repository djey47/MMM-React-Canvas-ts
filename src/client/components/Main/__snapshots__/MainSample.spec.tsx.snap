// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainSample component should render correctly with configuration context 1`] = `
<div
  className="main-sample dimmed light small"
>
  <p
    className="main-sample__description"
  >
    Main component sample for MM2's React-Typescript template
  </p>
  <p
    className="main-sample__configuration"
  >
    Configuration in context:
    <pre>
      <code>
        {"debug":true}
      </code>
    </pre>
  </p>
  <div
    prop1="text"
  >
    Subscribed component
  </div>
</div>
`;

exports[`MainSample component should render correctly with default props 1`] = `
<div
  className="main-sample dimmed light small"
>
  <p
    className="main-sample__description"
  >
    Main component sample for MM2's React-Typescript template
  </p>
  <p
    className="main-sample__configuration"
  >
    Configuration in context:
    <pre>
      <code>
        {}
      </code>
    </pre>
  </p>
  <div
    prop1="text"
  >
    Subscribed component
  </div>
</div>
`;
